### PR TITLE
Add additional free programming books (HTML, CSS, terminology)

### PR DESCRIPTION
**Pull request by j9t**
## What does this PR do?

Added several of my own titles that are either entirely free or whose source code is freely accessible.

### Description

Added the following titles:

* _CSS Optimization Basics_
* _Das kleine Buch der HTML-/CSS-Frameworks_ (German)
* _Rote Learning HTML & CSS_
* _Upgrade Your HTML_
* _The Web Development Glossary_

### Why is this valuable (or not)?

This should probably be answered by someone independent, however the titles cover topics relevant to web development practice, have been sold and/or read hundreds to thousands of times, and/or have been reviewed and published by major publishers (e.g., _Das kleine Buch der HTML-/CSS-Frameworks_ is the German version of the respective O’Reilly title).

### How do we know it's really free?

That may be best answered by the author, and I have no plans to make the repositories private. There’s a non-zero chance that could happen, but in that case, the titles could be forked if they haven’t been yet.

### For book lists, is it a book? For course lists, is it a course? etc.

It’s five books and they should be in the right places. (I hope, so not to update the PR again :)

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -971,12 +972,14 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Practical Series: A website template](https://practicalseries.com/1001-webdevelopment/) - Michael Gledhill (HTML)
 * [Pro HTML5 Programming](https://web.archive.org/web/20181215200026/http://apress.jensimmons.com/v5/pro-html5-programming/ch0.html) - Jen Simmons, Chris O'Connor, Dylan Wooters, Peter Lubbers *(:card_file_box: archived)*
 * [Resilient Web Design](https://resilientwebdesign.com/#Resilientweb%20design) - Jeremy Keith
+* [Rote Learning HTML & CSS](https://meiert.com/de/publications/books/rote-learning-html-and-css/) – Jens Oliver Meiert
```
Comment (eshellman) [inline books/free-programming-books-langs.md:975]: I think this should be on the German books page. Also, take a look at how we list other Leanpub books with 0 cost - still needs a login, and thus an access note.

**Review by eshellman — CHANGES_REQUESTED** [review]
nice additions. please see comment on rote learning

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (j9t) [reply books/free-programming-books-langs.md]: Good catch—corrected to go to the English landing page. The book is available on more platforms, including GitHub, and for free everywhere.

Comment (j9t) [reply books/free-programming-books-langs.md]: (Hope that also addresses the Leanpub point.)

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by j9t** [comment]
Thank you, @eshellman!

**Pull request merged**